src/packet.rs

```diff
@@ -467,7 +467,8 @@ pub mod login {
     pub mod clientbound { packets! {
         // 0x00 => Disconnect { reason: Chat }
         0x01 => EncryptionRequest { server_id: String, pubkey: Arr<Var<i32>, u8>, verify_token: Arr<Var<i32>, u8> }
-        // 0x02 => LoginSuccess { uuid: Uuid, username: String; impl Protocol for LoginSuccess { ... } } // NOTE: uuid field is encoded as a string!
+        // NOTE: uuid field is encoded as a string!
+        0x02 => LoginSuccess { uuid: String, username: String }
```

Comment:
And as I pointed out, this doesn't need to be a custom packet if we can [add a UuidString type](https://www.irccloud.com/pastebin/tpCJMXqh/uuidstring.rs) instead.